test_conformance/generic_address_space/advanced_tests.cpp

```diff
@@ -276,31 +304,24 @@ int test_library_function(cl_device_id deviceID, cl_context context, cl_command_
         NL "}"
         NL;
 
-    const std::string KERNEL_FUNCTION =
-        NL
-        NL "extern bool helperFunction(float *floatp, float val);"
-        NL
-        NL "__global float gfloat = 1.0f;"
-        NL
-        NL "__kernel void testKernel(__global uint *results) {"
-        NL "    uint tid = get_global_id(0);"
-        NL
-        NL "    __global float *gfloatp = &gfloat;"
-        NL "    __local float lfloat;"
-        NL "    lfloat = 2.0f;"
-        NL "    __local float *lfloatp = &lfloat;"
-        NL "    float pfloat = 3.0f;"
-        NL "    __private float *pfloatp = &pfloat;"
-        NL
-        NL "    uint failures = 0;"
-        NL
-        NL "    failures += helperFunction(gfloatp, gfloat) ? 0 : 1;"
-        NL "    failures += helperFunction(lfloatp, lfloat) ? 0 : 1;"
-        NL "    failures += helperFunction(pfloatp, pfloat) ? 0 : 1;"
-        NL
-        NL "    results[tid] = failures == 0;"
-        NL "}"
-        NL;
+    const std::string KERNEL_FUNCTION = NL NL
+        "extern bool helperFunction(float *floatp, float val);" NL NL
+        "#ifdef __opencl_c_program_scope_global_variables" NL
+        "__global float gfloat = 1.0f;" NL "#endif" NL NL
```

Comment:
The formatting changes don't make this very readable now.  I think you should be able to keep the original layout if you use a C++ raw string literal (and then you won't need `NL` any longer).

Reply:
I've made the change to raw string literally for this kernel.